File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        
        int c_len = matrix.length;
        int r_len = matrix[0].length;
        int[] row = new int[r_len];
        int[] col = new int[c_len];

        for(int i = 0; i < c_len; i++){
            for(int j =0; j < r_len; j++){
                if(matrix[i][j] == 0)
                {
                    row[j] = 1;
                    col[i] = 1;
                }
            }
        }
        for(int i = 0; i <  c_len; i++){
             if(col[i] == 1){
                for(int j = 0; j <  r_len; j++)              
                    matrix[i][j] = 0; 
             }
        }
        for(int i = 0; i <  r_len; i++){
            if(row[i] == 1){
                for(int j = 0; j <  c_len; j++)
                    matrix[j][i] = 0; 
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: list[list[int]]) -> None:
        clen = len(matrix)
        rlen = len(matrix[0])
        row = [0] * rlen
        col = [0] * clen 
        for i in range(clen):
            for j in range(rlen):
                if(matrix[i][j] == 0):
                    col[i] = 1
                    row[j] = 1
        
        for i in range(clen):
            if(col[i] == 1):
                for j in range(rlen):
                    matrix[i][j] = 0

        for i in range(rlen):
            if(row[i] == 1):
                for j in range(clen):
                    matrix[j][i] = 0 



        